diff --git a/src/game/SpellAuras.cpp b/src/game/SpellAuras.cpp
index a3b6e8b..ca4bf65 100644
--- a/src/game/SpellAuras.cpp
+++ b/src/game/SpellAuras.cpp
@@ -2657,9 +2657,9 @@ void Aura::HandleAuraModShapeshift(bool apply, bool Real)

                     if (m_modifier.m_miscvalue == FORM_CAT)
                     {
-                        m_target->SetPower(POWER_ENERGY,0);
-                        if(urand(1,100) <= FurorChance)
-                            m_target->CastSpell(m_target,17099,true,NULL,this);
+                        uint32 currEnergy = m_target->GetPower(POWER_ENERGY);
+                        uint32 newEnergy = (currEnergy <= FurorChance) ? currEnergy : FurorChance;
+                        m_target->SetPower(POWER_ENERGY,newEnergy);
                     }
                     else
                     {
diff --git a/src/game/Unit.cpp b/src/game/Unit.cpp
index fa47cd6..55ac971 100644
--- a/src/game/Unit.cpp
+++ b/src/game/Unit.cpp
@@ -6713,7 +6713,8 @@ void Unit::setPowerType(Powers new_powertype)
             break;
         case POWER_ENERGY:
             SetMaxPower(POWER_ENERGY,GetCreatePowers(POWER_ENERGY));
-            SetPower(   POWER_ENERGY,0);
+            if(GetTypeId()==TYPEID_PLAYER && this->getClass() != CLASS_DRUID)
+                SetPower(   POWER_ENERGY,0);
             break;
         case POWER_HAPPINESS:
             SetMaxPower(POWER_HAPPINESS,GetCreatePowers(POWER_HAPPINESS));